cli/index.js

```diff
@@ -251,28 +153,36 @@ exports.compileString = (sources, options) => {
       else argv.push(`--${key}`, String(val));
     }
   });
-  exports.main(argv.concat(Object.keys(sources)), {
-    stdout: output.stdout,
-    stderr: output.stderr,
+  return argv;
+}
+
+/** Convenience function that parses and compiles source strings directly. */
+export async function compileString(sources, config = {}) {
+  if (typeof sources === "string") sources = { [`input${defaultExtension.ext}`]: sources };
+  var argv = [
+    "--binaryFile", "binary",
+    "--textFile", "text",
+  ];
+  configToArguments(config, argv);
+  const output = {};
+  const result = await main(argv.concat(Object.keys(sources)), {
+    stdout: createMemoryStream(),
+    stderr: createMemoryStream(),
```

Comment:
It looks like `createMemoryStream()` uses quite often as default values for `stdout` and `stderr`. Could we predefine this as standard steams for `asc.main`?

Reply:
Becomes a little inconsistent in documentation then, since depending on where the CLI runs, Node or Web, the behavior is different, in that one cannot `toString` `process.stdout`. So specifying this explicitly avoids surprises.

Reply:
What we could do perhaps is invert this, so that a memory stream is always the default, and one would instead specify `process.stdout`. Effectively, the Node CLI is the only one doing that I think.

Reply:
Did that now :)